subsys/bluetooth/host/scan.c

```diff
@@ -606,6 +606,7 @@ void bt_hci_le_adv_ext_report(struct net_buf *buf)
 			 */
 			create_ext_adv_info(evt, &scan_info);
 			le_adv_recv(&evt->addr, &scan_info, &buf->b, evt->length);
+			(void)net_buf_pull_mem(buf, evt->length);
```

Comment:
I see we're adding that line to almost all branches. I think it'd be better to have it after L598 if possible. What do you think @LingaoM ?

Reply:
In fact, it will not be easy. This will modify `buf->data` in advance, resulting in subsequent logic changes.

Reply:
Use `goto cont` may better.